src/microbit/__model/accelerometer.py

```diff
@@ -0,0 +1,112 @@
+from . import constants as CONSTANTS
+
+
+class Accelerometer:
+    # The implementation is based off of https://microbit-micropython.readthedocs.io/en/v1.0.1/accelerometer.html.
+    def __init__(self):
+        self.__x = 0
+        self.__y = 0
+        self.__z = 0
+        self.__current_gesture = ""
+        self.__prev_gestures = set()
+        self.__gestures = []
+
+    def get_x(self):
+        """
+        Get the acceleration measurement in the ``x`` axis, as a positive or
+        negative integer, depending on the direction. The measurement is given in
+        milli-g.
+        """
+        return self.__x
+
+    def get_y(self):
+        """
+        Get the acceleration measurement in the ``y`` axis, as a positive or
+        negative integer, depending on the direction. The measurement is given in
+        milli-g.
+        """
+        return self.__y
+
+    def get_z(self):
+        """
+        Get the acceleration measurement in the ``z`` axis, as a positive or
+        negative integer, depending on the direction. The measurement is given in
+        milli-g.
+        """
+        return self.__z
+
+    def get_values(self):
+        """
+        Get the acceleration measurements in all axes at once, as a three-element
+        tuple of integers ordered as X, Y, Z.
+        """
+        return (self.__x, self.__y, self.__z)
+
+    def current_gesture(self):
+        """
+        Return the name of the current gesture.
+        """
+        self.__add_current_gesture_to_gesture_lists()
+        return self.__current_gesture
+
+    def is_gesture(self, name):
+        """
+        Return True or False to indicate if the named gesture is currently active.
+        """
+        self.__add_current_gesture_to_gesture_lists()
+        if name not in CONSTANTS.GESTURES:
+            raise ValueError(CONSTANTS.INVALID_GESTURE_ERR)
+        return name == self.__current_gesture
+
+    def was_gesture(self, name):
+        """
+        Return True or False to indicate if the named gesture was active since the
+        last call.
+        """
+        self.__add_current_gesture_to_gesture_lists()
+        if name not in CONSTANTS.GESTURES:
+            raise ValueError(CONSTANTS.INVALID_GESTURE_ERR)
+        was_gesture = name in self.__prev_gestures
+        self.__prev_gestures.clear()

```

Comment:
Is there a reason we are clearing this? The API doesn't seem to suggest that we lose the previous state

Reply:
prev_gestures is to remember all gestures that have existed since the last was_gesture call.
From the documentation, once the function is called, all the gestures in prev_gestures disappear (if that makes sense).